plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -30,9 +33,19 @@
         }
         return new DomainPushdownResult(domain, Domain.all(domain.getType()));
     };
-    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> new DomainPushdownResult(
-            domain.simplify(getDomainCompactionThreshold(session)),
-            domain);
+
+    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> {
+        checkArgument(
+                domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
+                "CASE_SENSITIVE_PUSHDOWN can be used only for chars and varchars");
+        if (domain.getValues().isDiscreteSet()) {
+            return new DomainPushdownResult(
+                    domain.simplify(getDomainCompactionThreshold(session)),
+                    domain);
+        }
+        return new DomainPushdownResult(Domain.all(domain.getType()), domain);
```

Comment:
This will help understand _what_ the code is doing:
```suggestion
return DISABLE_PUSHDOWN.apply(session, domain);
```

Add also a comment for _why_.